ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,226 @@
+
+-   Open the Software Centre via the start menu
+
+-   In the 'Applications' tab, click `Simba Spark ODBC Driver 64-bit`
+
+::: {align="center"}
+![](../images/databricks-software-centre.png)
+:::
+
+-   Click install
 
 ------------------------------------------------------------------------
 
-# Pre-requisites
+### Establishing an RStudio connection using environment variables
 
 ------------------------------------------------------------------------
 
-You must have:
+The `ODBC` package in RStudio allows you to connect to Databricks by creating and modifying four environment variables in your .Renviron file.
 
--   Access to Databricks
--   Access to a SQL Warehouse on DataBricks
--   R and RStudio downloaded
+To set the environment variables, call `usethis::edit_r_environ()`. You will then need to enter the following information:
+
+```         
+DATABRICKS_HOST=<databricks-host>
+DATABRICKS_SQL_WAREHOUSE_ID=<sql-warehouse-id>
+DATABRICKS_TOKEN=<personal-access-token> 
+DATABRICKS_CLUSTER_ID=<personal-cluster-id>
+```
+
+Once you have entered the details, save and close your .Renviron file.
+
+::: callout-note
+Everyone in your team that wishes to connect to the SQL Warehouse in Databricks and run your code must set up their .Renviron file individually, otherwise their connection will fail.
+:::
+
+The sections below describe where to find the information needed for each of the four environment variables.
 
 ------------------------------------------------------------------------
 
-# Downloading an ODBC driver
+#### Databricks host
 
 ------------------------------------------------------------------------
 
-1.  Install the 'Simba Spark ODBC' driver from the software centre.
+The Databricks host is the instance of Databricks that you want to connect to. It's the URL that you see in your browser bar when you're on the Databricks site and should end in "azuredatabricks.net" (ignore anything after this section of the URL).
 
-    i)  Open the Software Centre via the start menu.
+------------------------------------------------------------------------
 
-    ii) In the 'Applications' tab, click 'Simba Spark ODBC Driver 64-bit'. ![](../images/databricks-software-centre.png)
+#### Databricks SQL Warehouse ID
 
-    iii) Click install.
+------------------------------------------------------------------------
 
-2.  Get connection details for the SQL Warehouse from Databricks. To set up the connection you will need a few details from a SQL Warehouse within DataBricks.
+The Databricks SQL warehouse ID is the warehouse containing data that you would like to use. To get the warehouse ID, follow these steps:
 
-    i)  Login to [Databricks](https://adb-6882499576863257.17.azuredatabricks.net/?o=6882499576863257)
+-   click 'SQL Warehouses' under the 'SQL' section of the left hand menu on Databricks
+-   click on the warehouse name that you'd like to get the ID for
+-   the warehouse ID is in brackets next to the warehouse name on the 'Overview' tab
```

Comment:
This should be "the warehouse id is the 'HTTP Path' in the 'Connection details' tab". 

This makes the connection code a bit cleaner because otherwise the connection string has to append the `/sql/1.0/...`
```r
con <- DBI::dbConnect(
  odbc::databricks(),
  httpPath = paste0("/sql/1.0/warehouses/", Sys.getenv("DATABRICKS_SQL_WAREHOUSE_ID"))
)
```

This also future-proofs it a bit, because in theory they might change the connection details to be  `/sql/2.0/...` and that way only the environment variables have to change.

Reply:
Or just store the path as an environmental variable?

Reply:
Yep, this is what I'm suggesting:
```r
con <- DBI::dbConnect(
  odbc::databricks(),
  httpPath = Sys.getenv("DATABRICKS_SQL_WAREHOUSE_ID")
)

Sys.getenv("DATABRICKS_SQL_WAREHOUSE_ID")
[1] "/sql/1.0/warehouses/123abc..."
```

Reply:
This is a much easier way to explain it, thanks! The Databricks documentation wasn't super clear when I was digging around for ways to find each ID 😆